From 1893ae882ce8db51e3ce04bd628bf5ff4b6081b5 Mon Sep 17 00:00:00 2001
From: Francois Gaffie <francois.gaffie@intel.com>
Date: Fri, 23 Nov 2012 15:59:22 +0100
Subject: [PATCH 084/196] [PORT FROM MAIN] [AUDIO][RhB] Add Stream Flags key to setParameters

BZ: 70126

Audio Route Manager needs to know the type of stream to take the
appropriate routing decision (eg direct Output, MP3 Offload).
It must be define as key and filled by the policy.

Change-Id: I32b477180173cf779576ca5c2a07db732d2c9979
Origin-Change-Id: I6b5b2edc2fec2b8c4b3b2bc23281205ce71c760d
Signed-off-by: Francois Gaffie <francois.gaffie@intel.com>
Reviewed-on: http://android.intel.com:8080/78356
Reviewed-by: Barthes, FabienX <fabienx.barthes@intel.com>
Tested-by: Barthes, FabienX <fabienx.barthes@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 include/media/AudioParameter.h    |    2 ++
 media/libmedia/AudioParameter.cpp |    1 +
 2 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/include/media/AudioParameter.h b/include/media/AudioParameter.h
index 891bc4b..39e7bc1 100644
--- a/include/media/AudioParameter.h
+++ b/include/media/AudioParameter.h
@@ -41,6 +41,7 @@ public:
     //  keyInputSource: to change audio input source, value is an int in audio_source_t
     //     (defined in media/mediarecorder.h)
     //  keyScreenState: either "on" or "off"
+    //  keyStreamFlags: to give additional information to Audio HAL e.g. direct stream, offload
     static const char * const keyRouting;
     static const char * const keySamplingRate;
     static const char * const keyFormat;
@@ -48,6 +49,7 @@ public:
     static const char * const keyFrameCount;
     static const char * const keyInputSource;
     static const char * const keyScreenState;
+    static const char * const keyStreamFlags;
 
     String8 toString();
 
diff --git a/media/libmedia/AudioParameter.cpp b/media/libmedia/AudioParameter.cpp
index e3fea77..6fd497a 100644
--- a/media/libmedia/AudioParameter.cpp
+++ b/media/libmedia/AudioParameter.cpp
@@ -32,6 +32,7 @@ const char * const AudioParameter::keyChannels = AUDIO_PARAMETER_STREAM_CHANNELS
 const char * const AudioParameter::keyFrameCount = AUDIO_PARAMETER_STREAM_FRAME_COUNT;
 const char * const AudioParameter::keyInputSource = AUDIO_PARAMETER_STREAM_INPUT_SOURCE;
 const char * const AudioParameter::keyScreenState = AUDIO_PARAMETER_KEY_SCREEN_STATE;
+const char * const AudioParameter::keyStreamFlags = AUDIO_PARAMETER_KEY_STREAM_FLAGS;
 
 AudioParameter::AudioParameter(const String8& keyValuePairs)
 {
-- 
1.7.4.1

